main.js

```diff
@@ -0,0 +1,206 @@
+// basic extend function
+function extend(a, b) {
+  for (var key in b)
+    if (b.hasOwnProperty(key))
+      a[key] = b[key];
+  return a;
+}
+
+var DRAGnDROP = (function() {
+
+  var config = {
+    parentContainer: '.parent',
+    elementClass: 'drag-area',
+    handleFunction: log,
+  };
+
+  var element,
+    files;
+
+  function init(conf) {
+    if (conf) extend(config, conf);
+    element = document.createElement('div');
+    element.classList.add(config.elementClass);
+    element.addEventListener("dragenter", dragenter, false);
+    element.addEventListener("dragover", dragover, false);
+    element.addEventListener("drop", drop, false);
+    render();
+  }
+
+  // nothing to do on dragenter
+  function dragenter(e) {
+    e.stopPropagation();
+    e.preventDefault();
+  }
+
+  // nothing to do on dragover
+  function dragover(e) {
+    e.stopPropagation();
+    e.preventDefault();
+  }
+
+  // display thumbnails on drop event
+  // and creaate anchors for original files
+  function drop(e) {
+    e.stopPropagation();
+    e.preventDefault();
+    var dt = e.dataTransfer;
+    files = dt.files;
+    config.handleFunction(files);
+    // handleFiles(files); 
+  }
+
+  function log(fileList) {
+    console.log(fileList);
+  }
+
+  function render() {
+    var parent = document.querySelector(config.parentContainer);
+    parent.appendChild(element);
+  }
+
+  return {
+    init: init
+  };
+
+})();
+
+
+// gallery module
+var GALLERY = (function() {
+
+  // default configuration
+  var configuration = {
+    parentContainer: '#parent',
+    fileInput: '#files',
+    fileUploadButton: '#send',
+    dragArea: '#drag-area',
+    acceptedFormats: ['jpg', 'png'],
+    imageSize: {
+      width: 150,
+      height: 150
+    },
+    galleryArea: '#gallery'
+  };
+
+  // gallery init function
+  // overvrite defaults in config  
+  function init(config) {
+    if (config) extend(configuration, config);
+    start();
+  }
+
+  function start() {
+
+    var parent = document.querySelector(configuration.parentContainer);
+    var button = parent.querySelector(configuration.fileUploadButton);
+
+    // create file input field
+    var fileInput = document.createElement('input');
+    fileInput.type = 'file';
+    fileInput.name = 'files';
+    fileInput.accept = configuration.acceptedFormats.map(function(format) {
+      return '.' + format;
+    }).join(',');
+    fileInput.multiple = true;
+    parent.appendChild(fileInput);
+
+    // create upload button
+    var button = document.createElement('button');
+    button.id = configuration.fileUploadButton.replace('#', '');
+    button.name = 'send';
+    button.textContent = 'Upload';
+    parent.appendChild(button);
+
+    // create drag area container
+    DRAGnDROP.init({ parentContainer: configuration.parentContainer, handleFunction: handleFiles });
+
+    // create gallery container
+    var gallery = document.createElement('div');
```

Comment:
generalnie powinienes uniknac budowania elementow htmlowych w kodzie js. jest to duzo mniej konfigurowalne i daje duzo mniejsze mozliwosci manipulacji htmlem i warstwa prezentacji.